abi/abi_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package abi
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/codec"
+)
+
+func TestNewABI(t *testing.T) {
+	require := require.New(t)
+
+	actualABI, err := NewABI([]codec.Typed{
+		MockObjectSingleNumber{},
+		MockActionTransfer{},
+		MockObjectAllNumbers{},
+		MockObjectStringAndBytes{},
+		MockObjectArrays{},
+		MockActionWithTransfer{},
+		MockActionWithTransferArray{},
+		Outer{},
+	})
+	require.NoError(err)
+
+	expectedAbiJSON := mustReadFile(t, "testdata/abi.json")
+	expectedAbi := mustJSONParse[ABI](t, string(expectedAbiJSON))
```

Comment:
Could we update this to use `expectedABIJSON` and `expectedABI` ?

Reply:
What do you mean?